neuralogic/core/constructs/function/__init__.py

```diff
@@ -29,4 +32,78 @@
 Aggregation.SOFTMAX = Softmax("SOFTMAX")
 
 
-__all__ = ["Transformation", "Combination", "Aggregation", "Function"]
+class CombinationWrap:
+    __slots__ = "left", "right", "combination"
```

Comment:
Currently only works for binary operators. Consider extending this to support combinations on n elements.

Reply:
Extending this to n elements would mean supporting transformations out-of-the-box (???)